wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -0,0 +1,1614 @@
+                                                ConfigConstants.CONFIG_KEY_MERGE);
+            }
+
+            if (remoteBranch == null) {
+                remoteBranch = fullBranch;
+            }
+
+            FetchCommand fetchCommand = getGit().fetch();
+            fetchCommand.setRemote(remoteName);
+            if (fetchRefSpecs != null) {
+                fetchCommand.setRefSpecs(fetchRefSpecs);
+            }
+            int timeout = request.getTimeout();
+            if (timeout > 0) {
+                fetchCommand.setTimeout(timeout);
+            }
+
+            FetchResult fetchResult = (FetchResult)executeRemoteCommand(remoteUri, fetchCommand);
+
+            Ref remoteBranchRef = fetchResult.getAdvertisedRef(remoteBranch);
+            if (remoteBranchRef == null) {
+                remoteBranchRef = fetchResult.getAdvertisedRef(Constants.R_HEADS + remoteBranch);
+            }
+            if (remoteBranchRef == null) {
+                throw new GitException(String.format(ERROR_PULL_REF_MISSING, remoteBranch));
+            }
+            org.eclipse.jgit.api.MergeResult res = getGit().merge().include(remoteBranchRef).call();
+            if (res.getMergeStatus().equals(org.eclipse.jgit.api.MergeResult.MergeStatus.ALREADY_UP_TO_DATE)) {
+                return newDto(PullResponse.class).withCommandOutput("Already up-to-date");
+            }
+
+            if (res.getConflicts() != null) {
+                StringBuilder message = new StringBuilder(ERROR_PULL_MERGE_CONFLICT_IN_FILES);
+                message.append("\n");
+                Map<String, int[][]> allConflicts = res.getConflicts();
+                for (String path : allConflicts.keySet()) {
+                    message.append(path).append("\n");
+                }
+                message.append(ERROR_PULL_AUTO_MERGE_FAILED);
+                throw new GitException(message.toString());
+            }
+        } catch (CheckoutConflictException exception) {
+            StringBuilder message = new StringBuilder(ERROR_CHECKOUT_CONFLICT);
+            message.append("\n");
+            for (String path : exception.getConflictingPaths()) {
+                message.append(path).append("\n");
+            }
+            message.append(ERROR_PULL_COMMIT_BEFORE_MERGE);
+            throw new GitException(message.toString(), exception);
+        } catch (IOException | GitAPIException exception) {
+            String errorMessage;
+            if (exception.getMessage().equals("Invalid remote: " + remoteName)) {
+                errorMessage = ERROR_NO_REMOTE_REPOSITORY;
+            } else {
+                errorMessage = exception.getMessage();
+            }
+            throw new GitException(errorMessage, exception);
+        }
+        return newDto(PullResponse.class).withCommandOutput("Successfully pulled from " + remoteUri);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
```

Comment:
It is bad practise to write @SuppressWarnings for whole method. Can we locale this near variable that requires it?